File: linked_list/singly.go
package linked_list

import "fmt"

type Node struct {
	Data  int
	Child *Node
}

type LinkedList struct {
	Head *Node
}

func (list *LinkedList) Search(current *Node, number int) bool {
	if current.Data == number {
		return true
	} else if current.Child == nil {
		return false
	}
	return list.Search(current.Child, number)
}

func (list *LinkedList) Length() int {
	length := 1
	for current := list.Head; current.Child != nil; current = current.Child {
		length++
	}
	return length
}

func (list *LinkedList) Append(current *Node, node *Node) {
	if current.Child == nil {
		current.Child = node
		return
	}
	list.Append(current.Child, node)
}

func (list *LinkedList) Delete(current *Node, number int) {
	if current.Child.Data == number {
		temp := current.Child
		current.Child = current.Child.Child
		temp.Child = nil
		temp.Data = 0
		return
	}
	list.Delete(current.Child, number)
}

func (list *LinkedList) Contents() bool {
	var current *Node
	for current = list.Head; current.Child != nil; current = current.Child {
		fmt.Printf("%d, ", current.Data)
	}
	fmt.Println(current.Data)
	return true
}

func ReverseList(head *Node) *Node { // Iterative
	current := head
	var past *Node = nil
	var next *Node = nil

	for current != nil {
		next = current.Child
		current.Child = past
		past = current
		current = next
	}
	return past
}

File: linked_list/singly.py
class Node:
  def __init__(self, data):
    self.data = data
    self.child = None

class SinglyLinkedList:
  def __init__(self):
    self.head = None

  def insert(self, data):
    new_node = Node(data)
    new_node.child = self.head
    self.head = new_node

  def append(self, data):
    current = self.head
    while current.child is not None:
      current = current.child
    current.child = Node(data)

  def includes(self, data):
    current = self.head
    while current.data != data:
      if current.child is None:
        return False
      current = current.child
    return True

  def delete(self, data):
    current = self.head
    if current.data == data:
      head = current
      self.head = current.child
      head.child = None
      return
    while current.child is not None:
      if current.child.data == data:
        to_be_deleted = current.child
        current.child = current.child.child
        to_be_deleted.child = None
        return
      else: 
        current = current.child
    print(data, "not found")
  
  def reverse(self):
    prev = None
    current = self.head

    while current != None:
      next = current.child
      current.child = prev
      prev = current
      current = next
    self.head = prev
    
  def print_list(self):
    current = self.head
    while current is not None:
      print(current.data)
      current = current.child